staff/lab/mod-printer

```diff
@@ -40,40 +21,130 @@ def modify_printer(args):
         print('Action to be performed (and flag): %s (%s)' % (args.action, action_flag))
 
     # Call lpadmin command to perform addition/removal
-    for classname in cups_classes:
-        action_cmd = ['lpadmin', '-p', printer + '-' + classname, action_flag, classname]
+    for class_name, class_members in cups_classes.items():
+        printer_name = printer + '-' + class_name
+        if args.action == 'remove' and printer_name not in class_members:
+            # Allow the user to continue - supports removal if only one class was modified
+            print('ERROR: Printer %s is not a member of class %s' % (printer_name, class_name))
+            continue
+        action_cmd = ['lpadmin', '-p', printer_name, action_flag, class_name]
         subprocess.run(action_cmd, check=True)
         action_out = 'added to' if args.action == 'add' else 'removed from'
-        print('Printer %s was successfully %s %s' % (printer, action_out, classname))
+        print('Printer %s was successfully %s %s' % (printer, action_out, class_name))
 
 
 def list_printers(args):
     """Perform internal call to CUPS CLI to list printer status(es) (and potentially jobs)."""
-    # Initialize base lpstat command
-    lpstat_cmd = ['lpstat', '-c']
+    # Get available CUPS classes
+    lpstat_out = do_internal_call(['lpstat', '-c'])
+    cups_classes = parse_cups_classes(lpstat_out, args)
 
-    # Add on to lpstat command with arguments
-    if args.classname:
-        lpstat_cmd.append(args.classname)
-    if args.jobs:
-        lpstat_cmd.append('-o')
+    # Get all CUPS printers
+    lpstat_proc = subprocess.run(['lpstat', '-p'], check=True,
+                                 stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+    lpstat_out = lpstat_proc.stdout.decode('utf-8').strip().split('\n')
+
+    printer = args.printer and args.printer.lower()
+
+    # Parse lpstat output to find all available printers
+    # This includes classes and printers (i.e. shared and non-shared printers)
+    all_printers = []
+    for printer_line in lpstat_out:
+        printer_name_match = re.search(r'printer (\S*)', printer_line)
+        if not printer_name_match:
+            continue
+        printer_name = printer_name_match.group(1)
+
+        printer_arg = (printer and printer in printer_name) or not printer
+        class_arg = (args.classname and args.classname in printer_name) or not args.classname
```

Comment:
These two lines (and how they connect to the next if statement) are hard to understand. I'd recommend splitting it up something like this:

```py
if printer and printer not in printer_name:
  continue

if args.classname and args.classname not in printer_name:
  continue

if printer_name not in cups_classes:
  all_printers.append(printer_name)
```

This way it's easy to see where we skip and what the criteria it is that the code is looking for.